tests/functional/include/tests/config.h

```diff
@@ -4,21 +4,107 @@
 
 #pragma once
 
+#include "LogKit.h"
 #include "boost/ut.hpp"
 
 namespace ut = boost::ut;
 
 namespace cfg {
-struct printer : ut::printer {
-	template <class T>
-	auto operator<<(T &&t) -> auto &
+
+template <class TPrinter = ut::printer>
+class reporter : public ut::reporter<ut::printer>
+{
+  public:
+	auto on(ut::events::test_begin test_begin) -> void
+	{
+		// printer_ << '\n';
+		printer_ << "Running \"" << test_begin.name << "\"...";
+		fails_ = asserts_.fail;
+	}
+
+	auto on(ut::events::test_run test_run) -> void { printer_ << "\n \"" << test_run.name << "\"..."; }
+
+	auto on(ut::events::test_skip test_skip) -> void
+	{
+		printer_ << test_skip.name << "...SKIPPED\n";
+		++tests_.skip;
+	}
+
+	auto on(ut::events::test_end test_end) -> void
+	{
+		if (asserts_.fail > fails_) {
+			++tests_.fail;
+			printer_ << '\n';
+			printer_ << "Running \"" << test_end.name << "\"...";
+			printer_ << printer_.colors().fail << "FAILED" << printer_.colors().none << '\n';
+		} else {
+			++tests_.pass;
+			printer_ << '\n';
+			printer_ << "Running \"" << test_end.name << "\"...";
+			printer_ << printer_.colors().pass << "PASSED" << printer_.colors().none << '\n';
+		}
+	}
```

Comment:
The test is ended, it is not running anymore. Moreover previous format is different since it is on one line.

I suggest two things:

1. Do as default implementation by putting at the end `FAILED` or `PASSED`
2. As unit test rendering, replace Running by `OK`/`FAILED` or `Result of`

Reply:
```suggestion
	auto on(ut::events::test_end test_end) -> void
	{
		if (asserts_.fail > fails_) {
			++tests_.fail;
			printer_ << printer_.colors().fail << "FAILED" << printer_.colors().none << '\n';
		} else {
			++tests_.pass;
			printer_ << printer_.colors().pass << "PASSED" << printer_.colors().none << '\n';
		}
	}
```

<img width="1333" alt="Screenshot 2022-11-17 at 3 37 15 PM" src="https://user-images.githubusercontent.com/45776332/202478582-e48fa1d8-b888-41af-b30a-45aea23c88d3.png">

Reply:
```suggestion
	auto on(ut::events::test_end test_end) -> void
	{
		if (asserts_.fail > fails_) {
			++tests_.fail;
			printer_ << '\n';
			printer_ << "Result of \"" << test_end.name << "\"...";
			printer_ << printer_.colors().fail << "FAILED" << printer_.colors().none << '\n';
		} else {
			++tests_.pass;
			printer_ << '\n';
			printer_ << "Result of  \"" << test_end.name << "\"...";
			printer_ << printer_.colors().pass << "PASSED" << printer_.colors().none << '\n';
		}
	}
```

Example of unit test:

<img width="582" alt="Screenshot 2022-11-17 at 3 51 04 PM" src="https://user-images.githubusercontent.com/45776332/202478909-3c587e12-4931-450e-864e-7b2d7cc6fe23.png">